dataplane/kernel-forwarder/pkg/kernelforwarder/common.go

```diff
@@ -0,0 +1,220 @@
+		if err != nil {
+			logrus.Errorf("failed to get source namespace path - %v", err)
+			return nil, err
+		}
+		dstNsPath, err := crossConnect.GetLocalDestination().GetMechanism().NetNsFileName()
+		if err != nil {
+			logrus.Errorf("failed to get destination namespace path - %v", err)
+			return nil, err
+		}
+		return &connectionConfig{
+			srcNsPath: srcNsPath,
+			dstNsPath: dstNsPath,
+			srcName:   crossConnect.GetLocalSource().GetMechanism().GetParameters()[local.InterfaceNameKey],
+			dstName:   crossConnect.GetLocalDestination().GetMechanism().GetParameters()[local.InterfaceNameKey],
+			srcIP:     crossConnect.GetLocalSource().GetContext().GetIpContext().GetSrcIpAddr(),
+			dstIP:     crossConnect.GetLocalSource().GetContext().GetIpContext().GetDstIpAddr(),
+		}, nil
+	case cINCOMING:
+		dstNsPath, err := crossConnect.GetLocalDestination().GetMechanism().NetNsFileName()
+		if err != nil {
+			logrus.Errorf("failed to get destination namespace path - %v", err)
+			return nil, err
+		}
+		vni, _ := strconv.Atoi(crossConnect.GetRemoteSource().GetMechanism().GetParameters()[remote.VXLANVNI])
+		return &connectionConfig{
+			dstNsPath:  dstNsPath,
+			dstName:    crossConnect.GetLocalDestination().GetMechanism().GetParameters()[local.InterfaceNameKey],
+			dstIP:      crossConnect.GetLocalDestination().GetContext().GetIpContext().GetDstIpAddr(),
+			srcIPVXLAN: net.ParseIP(crossConnect.GetRemoteSource().GetMechanism().GetParameters()[remote.VXLANSrcIP]),
+			dstIPVXLAN: net.ParseIP(crossConnect.GetRemoteSource().GetMechanism().GetParameters()[remote.VXLANDstIP]),
+			vni:        vni,
+		}, nil
+	case cOUTGOING:
+		srcNsPath, err := crossConnect.GetLocalSource().GetMechanism().NetNsFileName()
+		if err != nil {
+			logrus.Errorf("failed to get destination namespace path - %v", err)
+			return nil, err
+		}
+		vni, _ := strconv.Atoi(crossConnect.GetRemoteDestination().GetMechanism().GetParameters()[remote.VXLANVNI])
+		return &connectionConfig{
+			srcNsPath:  srcNsPath,
+			srcName:    crossConnect.GetLocalSource().GetMechanism().GetParameters()[local.InterfaceNameKey],
+			srcIP:      crossConnect.GetLocalSource().GetContext().GetIpContext().GetSrcIpAddr(),
+			srcIPVXLAN: net.ParseIP(crossConnect.GetRemoteDestination().GetMechanism().GetParameters()[remote.VXLANSrcIP]),
+			dstIPVXLAN: net.ParseIP(crossConnect.GetRemoteDestination().GetMechanism().GetParameters()[remote.VXLANDstIP]),
+			vni:        vni,
+		}, nil
+	default:
+		logrus.Error("connection configuration: invalid connection type")
+		return nil, fmt.Errorf("invalid connection type")
+	}
+}
+
+func newVETH(srcName, dstName string) *netlink.Veth {
+	/* Populate the VETH interface configuration */
+	return &netlink.Veth{
+		LinkAttrs: netlink.LinkAttrs{
+			Name: srcName,
+			MTU:  cVETHMTU,
+		},
+		PeerName: dstName,
+	}
+}
+
+func newVXLAN(ifaceName string, egressIP, remoteIP net.IP, vni int) *netlink.Vxlan {
```

Comment:
That is simple enough, but belongs to `remote.go`. Same for `newVETH` which belogs to `local.go`.

Reply:
Moved it 👍